packages/listview/src/components/ColoredStatusColumn/index.js

```diff
@@ -1,30 +1,95 @@
-import getNormalizedColors from './helpers/getNormalizedColors';
 import getColor from './helpers/getColor';
-import { StyledContainer } from './styled';
+import getNormalizedColors from './helpers/getNormalizedColors';
+import useDefaultColors from './hooks/useDefaultColors';
+import resolvePosition from './helpers/resolvePosition';
+import { StyledContainer, StyledDropdown, StyledIndicator } from './styled';
+
+const ColoredStatusColumn = ({ row, value, colors, textTransform, isEditable, onChange }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const containerRef = useRef();
 
-const ColoredStatusColumn = ({ value, colors, textTransform }) => {
     const defaultColors = useDefaultColors();
     const map = colors ? getNormalizedColors(colors) : defaultColors;
     const backgroundColor = getBackgroundColor(map[value.toLowerCase()]);
     const color = getColor(map[value.toLowerCase()]);
+    const hasPopup = isEditable ? 'listbox' : null;
+
+    const handleChange = (value) => {
+        const { name } = value;
+        onChange({
+            row,
+            value: name,
+        });
+    };
+
+    const handleContainerClick = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const positionResolver = useCallback((opts) => resolvePosition(opts, 'right'), []);
+    useWindowResize(() => setIsOpen(false), isOpen);
+    useOutsideClick(
+        containerRef,
+        (event) => {
+            if (event.target !== containerRef.current) {
+                setIsOpen(false);
+            }
+        },
+        isOpen,
+    );
+
+    const getOptions = () =>
+        Object.keys(map).map((value) => {
+            const color = getColor(map[value.toLowerCase()]);
+            const backgroundColor = getBackgroundColor(map[value.toLowerCase()]);
+            return (
+                <Option
+                    name={value}
+                    label={value}
+                    color={color}
+                    backgroundColor={backgroundColor}
+                    textTransform={textTransform}
+                    component={ColoredOption}
+                />
+            );
+        });
 
     return (
         <StyledContainer
             backgroundColor={backgroundColor}
             color={color}
             textTransform={textTransform}
+            ref={containerRef}
+            onClick={handleContainerClick}
+            aria-haspopup={hasPopup}
+            aria-expanded={isOpen}
+            isEditable
         >
             {value}
+            <RenderIf isTrue={isEditable}>
+                <StyledIndicator color={color} />
+                <InternalOverlay
+                    isVisible={isOpen}
+                    positionResolver={positionResolver}
+                    triggerElementRef={() => containerRef}
+                >
+                    <StyledDropdown value={{ name: value }} onChange={handleChange}>
+                        <HeaderOption />
+                        {getOptions()}
```

Comment:
let's use this declarative:
```<Options />```
create Options components that make the iteration